libs/java/server_common/src/main/java/com/yahoo/athenz/common/server/util/config/ConfigManager.java

```diff
@@ -0,0 +1,463 @@
+/*
+ * Copyright 2017 Yahoo Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.yahoo.athenz.common.server.util.config;
+
+import com.yahoo.athenz.common.server.util.Utils;
+import com.yahoo.athenz.common.server.util.config.dynamic.DynamicConfigDuration;
+import com.yahoo.athenz.common.server.util.config.providers.ConfigProvider;
+import com.yahoo.athenz.common.server.util.config.providers.ConfigProviderAwsParametersStore;
+import com.yahoo.athenz.common.server.util.config.providers.ConfigProviderFile;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.io.Closeable;
+import java.lang.invoke.MethodHandles;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.TimeUnit;
+
+/**
```

Comment:
I think explaining how it works in comments is unnecessary. The best place to put this info is in a design doc.
Whoever needs to change it in the future will still look at the code and if he'll change some functionality I doubt he'll remember to update the comments.

Reply:
I disagree. The comment doesn't explain the internals - but rather how to use the package.
We'll let others judge.